backend/graphql/resolvers/userResolvers.ts

```diff
@@ -63,13 +82,27 @@ const userResolvers = {
       _parent: undefined,
       { id }: { id: string },
     ): Promise<void> => {
-      return userService.deleteUserById(id);
+      const teacher = await userService.getUserById(id);
+      const classes = teacher.class;
+      userService.deleteUserById(id);
+      if (classes) {
+        classes.forEach((classId) => {
+          classService.deleteClass(classId);
+        });
+      }
     },
     deleteUserByEmail: async (
       _parent: undefined,
       { email }: { email: string },
     ): Promise<void> => {
-      return userService.deleteUserByEmail(email);
+      const teacher = await userService.getUserByEmail(email);
+      const classes = teacher.class;
+      userService.deleteUserByEmail(email);
+      if (classes) {
+        classes.forEach((classId) => {
+          classService.deleteClass(classId);
+        });
+      }
     },
   },
 };
```

Comment:
so, mongo has built-in functionality for removing dependent documents. you can see an example of how to use this in a similar `[PR](https://github.com/uwblueprint/jump-math/pull/145/files)` made by carissa! documentation for this as well [here](https://mongoosejs.com/docs/middleware.html#pre).

the reason for this is that if the class deletion fails, we can prevent the teacher deletion from happening and throw an error instead (so that there are no orphaned class objects in our db!) cc @carissa-tang who might have more context here

Reply:
I see that makes a lot more sense - I have made the changes!